src/app/containers/ContactsContainer.js

```diff
@@ -161,11 +162,18 @@ const ContactsContainer = ({ location, history }) => {
                 </ConfirmModal>
             );
         });
-        await api(checkAll && !contactGroupID ? clearContacts() : deleteContacts(activeIDs));
-        history.replace('/contacts');
+        if (checkAll && !contactGroupID) {
+            await api(clearContacts());
+            return createNotification({ text: c('Success').t`Contacts deleted` });
+        }
+        const { Responses } = await api(deleteContacts(activeIDs));
+        history.push({ ...location, pathname: '/contacts' });
         await call();
         setCheckedContacts(Object.create(null));
         setCheckAll(false);
+        if (Responses.some(({ Response: { Code } }) => Code !== SUCCESS_API_CODE)) {
```

Comment:
We could move this checking in a `proton-shared` api helper.

Reply:
I'm gonna check if the API routes for array submission always have the structure:
```
{
  "Code": 1001,
  "Responses": [
    {
      "Response": {
        "Code": 1000
      },
      ...
    },
    ...
  ]
}
```
At least for `contacts` routes, that's the case

Reply:
@bartbutler can tell you.

Reply:
I browsed through the api specs and it does seem to be the case